tests/test_sig_stfl.c

```diff
@@ -0,0 +1,417 @@
+		secret_key->save_secret_key = do_nothing_save;
+	}
+
+	if (secret_key == NULL) {
+		fprintf(stderr, "ERROR: OQS_SECRET_KEY_new failed\n");
+		goto err;
+	}
+
+	/**
+	 * Magic numbers are not inserted into the secret key because for we need an
+	 * OQS_SECRET_KEY object for the secret key. The secret key byte array is an attribute
+	 * of the object whose size gets assigend when new() is called.
+	 */
+
+	// Set the magic numbers before
+	memcpy(public_key, magic.val, sizeof(magic_t));
+	memcpy(message, magic.val, sizeof(magic_t));
+	memcpy(signature, magic.val, sizeof(magic_t));
+
+	public_key += sizeof(magic_t);
+	message += sizeof(magic_t);
+	signature += sizeof(magic_t);
+
+	// and after
+	memcpy(public_key + sig->length_public_key, magic.val, sizeof(magic_t));
+	memcpy(message + message_len, magic.val, sizeof(magic_t));
+	memcpy(signature + sig->length_signature, magic.val, sizeof(magic_t));
+
+	// =================================== KEYPAIR / FILE READ =========================================
+
+	OQS_randombytes(message, message_len);
+	OQS_TEST_CT_DECLASSIFY(message, message_len);
+
+	if (mode == READING) {
+		strcpy(filename, filestem);
+
+		// Public Key
+		strcat(filename, PUBLIC_KEY_EXT);
+		FILE *pub_key = fopen(filename, "rb");
+		if (pub_key == NULL) {
+			goto err;
+		}
+
+		size_t result = fread(public_key, 1, sig->length_public_key, pub_key);
+		if (result != sig->length_public_key) {
+			goto err;
+		}
+		fclose(pub_key);
+
+		// Private Key
+		filename[strlen(filename) - strlen(PUBLIC_KEY_EXT)] = '\0';
+		strcat(filename, PRIVATE_KEY_EXT);
+		FILE *prv_key = fopen(filename, "rb");
+		if (prv_key == NULL) {
+			goto err;
+		}
+
+		result = fread(secret_key->secret_key, 1, secret_key->length_secret_key, prv_key);
+		if (result != secret_key->length_secret_key) {
+			goto err;
+		}
+		fclose(prv_key);
+
+	} else {
+		rc = OQS_SIG_STFL_keypair(sig, public_key, secret_key);
+	}
+	OQS_TEST_CT_DECLASSIFY(&rc, sizeof rc);
+	if (rc != OQS_SUCCESS) {
+		if (mode == READING) {
+			fprintf(stderr, "ERROR: OQS_SIG_STFL read from file failed\n");
+		} else {
+			fprintf(stderr, "ERROR: OQS_SIG_STFL_keypair failed\n");
+		}
+		goto err;
+	}
+
+	// Save the keypair to the specified filestem
+	if (mode == SAVING) {
+		// Private Key
+		secret_key->save_secret_key(secret_key);
+
+		// Public Key
+		strcpy(filename, filestem);
+		strcat(filename, PUBLIC_KEY_EXT);
+		FILE *pub_key = fopen(filename, "w+");
+		if (pub_key == NULL) {
+			fprintf(stderr, "ERROR: fopen failed.\n");
+			goto err;
+		}
+
+		for (unsigned int i = 0; i < sig->length_public_key; i++) {
+			if (fputc(public_key[i], pub_key) == EOF) {
+				goto err;
+			}
+		}
+		fclose(pub_key);
+	}
+
+	// ===========================================================================================
+
+
+	// ====================================== SIGNING =============================================
+
+	rc = OQS_SIG_STFL_sign(sig, signature, &signature_len, message, message_len, secret_key);
```

Comment:
Just for debugging purposes, could check here to be sure that the output `signature_len` is no bigger than the maximum signature length we allocated.